src/Http/Headers/src/CookieHeaderParser.cs

```diff
@@ -57,7 +61,65 @@ public override bool TryParseValue(StringSegment value, ref int index, out Cooki
             }
 
             index = current;
-            parsedValue = result;
+
+            return true;
+        }
+
+        public bool TryParseValues(StringValues values, Dictionary<string, string> store, bool enableCookieNameEncoding)
+        {
+            // If a parser returns an empty list, it means there was no value, but that's valid (e.g. "Accept: "). The caller
+            // can ignore the value.
+            if (values.Count == 0)
+            {
+                return false;
+            }
+            bool hasFoundValue = false;
+
+            for (var i = 0; i < values.Count; i++)
+            {
+                var value = values[i];
+                var index = 0;
+
+                while (!string.IsNullOrEmpty(value) && index < value.Length)
+                {
+                    if (TryParseValue(value, ref index, out var parsedName, out var parsedValue))
+                    {
+                        // The entry may not contain an actual value, like " , "
+                        if (parsedName != null && parsedValue != null)
+                        {
+                            var name = enableCookieNameEncoding ? Uri.UnescapeDataString(parsedName.Value.Value) : parsedName.Value.Value;
```

Comment:
You can end up calling `Uri.UnescapeDataString(parsedName.Value.Value)` twice, might want to add an `if` to avoid that.

Reply:
I don't think it would call Uri.UnescapeDataString on the `parsedName` twice here, unless I'm missing something about ternaries

Reply:
1. `enableCookieNameEncoding ? Uri.UnescapeDataString(parsedName.Value.Value) : parsedName.Value.Value;`
2. `store[name] = Uri.UnescapeDataString(parsedValue.Value.Value);`

Reply:
One is on `parsedName` and the other on `parsedValue`

Reply:
Oh wow, I'm blind :D